Source/SwiftLintFramework/Rules/RuleConfigurations/DiscouragedInitConfiguration.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  DiscouragedInitConfiguration.swift
+//  SwiftLint
+//
+//  Created by Ornithologist Coder on 8/1/17.
+//  Copyright © 2017 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct DiscouragedInitConfiguration: RuleConfiguration, Equatable {
+    public var severityConfiguration = SeverityConfiguration(.warning)
+
+    public var consoleDescription: String {
+        return severityConfiguration.consoleDescription
+    }
+
+    public var severity: ViolationSeverity {
+        return severityConfiguration.severity
+    }
+
+    private(set) public var discouragedInits: [String]
+
+    private let defaultDiscouragedInits = [
+        "Bundle",
+        "UIDevice"
+    ]
+
+    init() {
+        discouragedInits = defaultDiscouragedInits
+    }
+
+    // MARK: - RuleConfiguration
+
+    public mutating func apply(configuration: Any) throws {
+        guard let configuration = configuration as? [String: Any] else {
+            throw ConfigurationError.unknownConfiguration
+        }
+
+        if let severityString = configuration["severity"] as? String {
+            try severityConfiguration.apply(configuration: severityString)
+        }
+
+        if let included = [String].array(of: configuration["included"]) {
+            discouragedInits += included
```

Comment:
I think this should always override the default values. Doing that, we provide a way to opt-out from the default configuration if needed (for example, one might have a class `Bundle` in their project).